src/clients/AcrossConfigStoreClient.ts

```diff
@@ -0,0 +1,441 @@
+  sortEventsAscending,
+} from "../utils";
+
+import { Contract, BigNumber } from "ethers";
+import winston from "winston";
+import { Block } from "@ethersproject/abstract-provider";
+
+import {
+  L1TokenTransferThreshold,
+  TokenConfig,
+  GlobalConfigUpdate,
+  ParsedTokenConfig,
+  SpokeTargetBalanceUpdate,
+  SpokePoolTargetBalance,
+  RouteRateModelUpdate,
+  ConfigStoreVersionUpdate,
+  DisabledChainsUpdate,
+} from "../interfaces";
+
+import * as lpFeeCalculator from "../lpFeeCalculator";
+import { BlockFinder, across } from "@uma/sdk";
+import { HubPoolClient } from "./HubPoolClient";
+
+export const GLOBAL_CONFIG_STORE_KEYS = {
+  MAX_RELAYER_REPAYMENT_LEAF_SIZE: "MAX_RELAYER_REPAYMENT_LEAF_SIZE",
+  MAX_POOL_REBALANCE_LEAF_SIZE: "MAX_POOL_REBALANCE_LEAF_SIZE",
+  VERSION: "VERSION",
+  DISABLED_CHAINS: "DISABLED_CHAINS",
+};
+
+export class AcrossConfigStoreClient {
+  public readonly blockFinder: BlockFinder<Block>;
+
+  public cumulativeRateModelUpdates: across.rateModel.RateModelEvent[] = [];
+  public cumulativeRouteRateModelUpdates: RouteRateModelUpdate[] = [];
+  public cumulativeTokenTransferUpdates: L1TokenTransferThreshold[] = [];
+  public cumulativeMaxRefundCountUpdates: GlobalConfigUpdate[] = [];
+  public cumulativeMaxL1TokenCountUpdates: GlobalConfigUpdate[] = [];
+  public cumulativeSpokeTargetBalanceUpdates: SpokeTargetBalanceUpdate[] = [];
+  public cumulativeConfigStoreVersionUpdates: ConfigStoreVersionUpdate[] = [];
+  public cumulativeDisabledChainUpdates: DisabledChainsUpdate[] = [];
+
+  private rateModelDictionary: across.rateModel.RateModelDictionary;
+  public firstBlockToSearch: number;
+
+  public hasLatestConfigStoreVersion = false;
+
+  public isUpdated = false;
+
+  constructor(
+    readonly logger: winston.Logger,
+    readonly configStore: Contract,
+    readonly hubPoolClient: HubPoolClient,
+    readonly eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 },
+    protected readonly configOverride: {
+      enabledChainIds: number[];
+      defaultConfigStoreVersion: number;
+    } = { enabledChainIds: [], defaultConfigStoreVersion: 0 }
+  ) {
+    this.firstBlockToSearch = eventSearchConfig.fromBlock;
+    this.blockFinder = new BlockFinder(this.configStore.provider.getBlock.bind(this.configStore.provider));
+    this.rateModelDictionary = new across.rateModel.RateModelDictionary();
+  }
+
+  async computeRealizedLpFeePct(
+    deposit: { quoteTimestamp: number; amount: BigNumber; destinationChainId: number; originChainId: number },
+    l1Token: string
+  ): Promise<{ realizedLpFeePct: BigNumber; quoteBlock: number }> {
+    let quoteBlock = await this.getBlockNumber(deposit.quoteTimestamp);
+
+    if (!quoteBlock) {
+      throw new Error(`Could not find block for timestamp ${deposit.quoteTimestamp}`);
+    }
+
+    // There is one deposit on optimism for DAI that is right before the DAI rate model was added.
+    if (quoteBlock === 14830339) {
```

Comment:
Just realized that we can remove these hardcoded catches from line 85-97 now that we have new spoke pools.

For Goerli, we'll just redeploy the spoke pool